### PR TITLE
Allow for env-var substitution inside query_string and table_name

### LINKED ISSUE #94: Allow for env-var substitution inside query_string and table_name
For some use cases, the exact query string or target table name may differ from run to run (e.g. we are looking to sync all order rewards for the current month into a dune table that is suffixed with the current month).

I don't think it is necessary to support a sophisticated templating syntax for this. 

However, I think it would be nice if I could specify a config like:

```yaml
- name: order_rewards
  source:
   ref: postgres
   query_string: "SELECT * from order_rewards where DATE_TRUNC('month', timestamp) = '${DATE_RANGE}'"
  destination:
    ref: dune
    table_name: order_rewards_${DATE_RANGE}
```

and then if `DATE_RANGE=2024-11-01` in the environment we fetch order rewards from November and upload them into `order_rewards_2024-11-01`

### PR DESCRIPTION
**Pull request by mooster531**
Closes #94

### CODE REVIEW

**Review by bh2smith — COMMENTED** [review]
This looks alright, Just some extra thoughts to simplify. 

We might as well just do the same for postgres. Also, couldn't we extrapolate the env at config loading time so we don't have to import this in all the constructors? This would make it more universal for adding to future sources. Just template all query strings and table names on all sources and destinations. 

Actually if we went this route we wouldn't need to add so many tests to the database classes.

**Comment by mooster531** [comment]
If we're aiming to support environment variable substitution anywhere and everywhere in the entire config file, then I suggest we remove our `interpolate` magic altogether and advise the users to take advantage of [`envsubst`](https://www.gnu.org/software/gettext/manual/html_node/envsubst-Invocation.html).

The execution environment of the program is entirely static anyway - you run a specific docker image with a specific previously-prepared config file in all cases.

**Comment by mooster531** [comment]
Draft-ing this until I incorporate the suggestion to support env interpolate in the entire config.yaml

File: src/config.py [hunk 1]
```diff
@@ -117,47 +116,6 @@ def interpolate(value: str) -> str:
             raise KeyError(f"Environment variable '{missing_var}' not found. ") from e
 
 
-def parse_query_parameters(params: list[dict[str, Any]]) -> list[QueryParameter]:
```
Comment (bh2smith) [inline src/config.py:120]: Its cool that this is moved, but Its only used in this file and its part of the config parsing. Moving this makes sense, however now I would argue that it should become a private static method on the DuneSource class (a constructor helper) used to parse the `param: dict[list[str, Any]]`.

File: src/sources/dune.py [hunk 2]
```diff
@@ -1,5 +1,7 @@
 """Source logic for Dune Analytics."""
 
+from __future__ import annotations
```
Comment (bh2smith) [inline src/sources/dune.py:3]: What was this for?

Comment (mooster531) [reply src/sources/dune.py:3]: Honestly, no idea. It was already there, or potentially added by one of our very clever linters/fixers.

**Review by bh2smith — APPROVED** [review]
Btw. I approve. The query parameter parsing can be done in a follow up.

File: src/config.py (hunk 1 above)
Comment (mooster531) [reply src/config.py:120]: Then the same would apply to the following functions in `src/sources/dune.py`:

```
_parse_decimal_type
_reformat_varbinary_columns
_reformat_unknown_columns
_handle_column_types
dune_result_to_df
```

Comment (bh2smith) [reply src/config.py:120]: It doesn't have to be a private static class method.

**Pull request merged**